src/Raven.Server/Documents/Sharding/Handlers/ShardedIncomingReplicationHandler.cs

```diff
@@ -279,7 +330,48 @@ protected override void DisposeInternal()
                 }
             });
 
+            if (_batches != null)
+            {
+                _batches.DisposeStreams = true;
+                _batches.Dispose();
```

Comment:
can `_batches` disposed concurrently?  from the `using` & from disposing the handler?

Reply:
This worries me TBH... can we not reuse the `_batches` ?

Reply:
Done